src/executor.c

```diff
@@ -158,14 +158,36 @@ t_err	ft_process_cmd(t_cmd *cmd, t_err err, t_data *data)
 {
 	if (err == ERR_MALLOC)
 		return (err);
+	else if (err == ERR_DIR)
+		return (SUCCESS);
 	else if (err == ERR_UNKNOWN_CMD)
 	{
-		write(2, "minishell: ", 11);
-		ft_putstr_fd(cmd->args[0], 2);
-		write(2, ": command not found\n", 20);
+		ft_print_warning("nocmd", cmd->args[0]);
 		err = ft_close(&cmd->fd_pipe[1]);
 	}
 	else if (err == SUCCESS)
 		err = ft_create_child(cmd, data, false);
 	return (err);
```

Comment:
This returns with SUCCESS, which means wait_for_children also tries to wait for empty commands.

My guess: In the if condition of waitpid also add (cmd->pid != 0 &&)

Reply:
What do you mean, that ft_print_warning() returns SUCCESS when "nocmd"?

That's why I don't catch the err of it in that case. But I see the err of ft_close will be SUCCESS usually. Solved if temporarily, but should be adapted with the refactor mentioned above.

cmd->pid initialised to 0